JASP-Engine/JASP/R/descriptives.R

```diff
@@ -111,6 +111,45 @@ Descriptives <- function(jaspResults, dataset, options) {
       }
     }
   }
+
+  # QQ plots
+  if (options$descriptivesQQPlot) {
+    if(is.null(jaspResults[["QQPlots"]])) {
+      if(length(variables)>1 || length(levels(dataset[[.v(splitName)]]))>1) #there will be more than one Q-Q Plot
+        jaspResults[["QQPlots"]] <- createJaspContainer("Q-Q Plots")
+      else #only one Q-Q Plot
+        jaspResults[["QQPlots"]] <- createJaspContainer("Q-Q Plot")
+      jaspResults[["QQPlots"]]$dependOn(c("descriptivesQQPlot", "splitby"))
+      jaspResults[["QQPlots"]]$position <- 8
+    }
+    QQPlots <- jaspResults[["QQPlots"]]
+    if(makeSplit) {
+      qqSplitFactor     <- dataset[[.v(splitName)]]
+      if(length(qqSplitFactor)==0)
+        return(createJaspPlot(error="Plotting is not possible: Variable only contains NA!", dependencies="splitby"))
+      #gives the different split values
+      qqSplitLevels     <- levels(qqSplitFactor)
+      # remove missing values from the grouping variable
+      dataset           <- dataset[!is.na(qqSplitFactor), ]
+      for(var in variables){ 
+        deeperQQPlots <- createJaspContainer(paste0(var))
+        deeperQQPlots$dependOn(optionContainsValue=list(variables=var))
```

Comment:
Maybe I'm missing it but are you checking if the plot was already created?
Because right now I think it still does the plotting again, only you don't see it because it doesn't remove the prior plot until the new one is created again.